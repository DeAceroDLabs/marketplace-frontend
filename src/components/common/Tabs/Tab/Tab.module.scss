@use "styles/colors";
@use "styles/typography";

.tab {
  cursor: pointer;
  margin: 0 12px;
}

.secondary {
  min-width: 197px;
  height: 42px;
  background: #ffffff;
  border: 1px solid colors.$accent;
  color: colors.$accent;
  border-radius: 80px;
}

.secondary.active {
  background-color: colors.$accent;
  color: colors.$neutral-inverse;
}
