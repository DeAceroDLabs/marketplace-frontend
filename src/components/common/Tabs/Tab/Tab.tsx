import { TabItem } from "components/common/common.types";
import styles from "./Tab.module.scss";

export interface TabProps {
  tab: TabItem;
  variant: "primary" | "secondary";
  active: "active" | "";
  onSelect: (tabId: TabItem) => void;
}

const Tab: React.FC<TabProps> = ({ tab, variant, active, onSelect }) => {
  const setActive = () => onSelect(tab);
  return (
    <button
      className={`${styles.tab} ${styles[variant]} ${styles[active]}`}
      onClick={() => setActive()}
    >
      {tab.title}
    </button>
  );
};

export default Tab;
