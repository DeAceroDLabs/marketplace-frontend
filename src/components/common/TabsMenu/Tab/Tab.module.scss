@use 'styles/colors';
@use 'styles/typography';

.button {
  cursor: pointer;
}

.secondary {
  width: 197px;
  height: 42px;
  background: #FFFFFF;
  border: 1px solid colors.$accent;
  color: colors.$accent;
  border-radius: 80px;
}

.secondary.active {
  background-color: colors.$accent;
  color: colors.$neutral-inverse;
}
