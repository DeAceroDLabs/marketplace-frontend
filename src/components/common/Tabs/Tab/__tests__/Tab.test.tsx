import renderer from "react-test-renderer";
import Tab, { TabProps } from "../Tab";
import { TabItem } from "../../../common.types";

const mockTab: TabItem = {
  id: 1,
  title: "tab 1",
};

const TabMockProps: TabProps = {
  tab: mockTab,
  active: "active",
  variant: "primary",
  onSelect: () => jest.fn(),
};

it("renders Tab with no issue", () => {
  const component = renderer.create(<Tab {...TabMockProps} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
