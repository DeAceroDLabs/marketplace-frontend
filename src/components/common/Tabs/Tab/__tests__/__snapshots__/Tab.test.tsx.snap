// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Tab with no issue 1`] = `
<button
  className="tab primary active"
  onClick={[Function]}
>
  tab 1
</button>
`;
